calculating_n50_assemblies.py

```diff
@@ -3,60 +3,81 @@
 import operator
 import sys
 import os
+import statistics
+import math
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
 
 def main(in_dir = None, out_dir = None):
     if os.path.exists(in_dir) == False: 
         sys.exit("Input directory or path incorrect. Exiting..")
     if os.path.exists(out_dir) == False: 
         os.mkdir(out_dir)
-    output_file = open(os.path.join(out_dir,'N50.txt'),'w')
+    output_file = open(os.path.join(out_dir,'summary_statistics.txt'),'w')
     fasta_ext = ['fasta','fna','ffn','faa','frn']
-
+    n50_array = [] 
+    n50_array_log = []
     switch = 'off'
     for f_name in os.listdir(in_dir):
         if f_name.endswith('.gz'):
             array_f = f_name.split('.')
             if array_f[-2] in fasta_ext:
                 file = os.path.join(in_dir,f_name)
                 with gzip.open(file, mode = 'rt') as open_file:
-                    n50_calc(open_file, output_file)
+                    n50_calc(open_file, n50_array, n50_array_log)
                 switch = 'on'
         else:
             array_f = f_name.split('.')
             if array_f[-1] in fasta_ext:
                 file = os.path.join(in_dir,f_name)
                 with open(file, mode = 'r') as open_file:
-                    n50_calc(open_file, output_file)
+                    n50_calc(open_file, n50_array, n50_array_log)
                 switch = 'on'
+    
+    if switch == 'on':
```

Comment:
Could you think of a way of maintaining your application flow without a switch variable?